### DISCUSSION TITLE
Logging diarization progress

### DISCUSSION THREAD

**Discussion by ataibarkai** [pyannote/pyannote-audio#1160] [Q&A]
Is there a way to log progress of the diarization pipeline?

For context, the motivation: some diarization attempts end up either taking much much longer than others, or hanging altogether.
Knowing how diarization is progressing is the first step in debugging this.

Ideally, I'd like a callback getting called, so I can use custom logging when needed.

Thank you!

**Comment by hbredin** [comment]
There is no such thing. I'd gladly have a look at a PR adding this feature.
It has been discussed [here](https://github.com/audapolis/audapolis/issues/366#issuecomment-1144937656) as well.
